### ISSUE TITLE
Uncaught SyntaxError: Unexpected end of input 

### ISSUE THREAD

**Issue by slorber** [milankinen/livereactload#35]
I tried to use the API as I manage all the state outside of React.

```
LiveReactLoadApi.onReload(function() {
    console.error("test");
});
```

The first time I edit a JSX it prints "test" and then it prints this error message:
`Uncaught SyntaxError: Unexpected end of input `

It seems to be on require lines like:
```
module.exports = GlobeIcon;
;require("livereactload/lib/browser/reloadify-exports")("/home/sebastien/Desktop/Stample-react/src/img/icons/globe.jsx", module);
```

or

```
;require("livereactload/lib/browser/reloadify-exports")("/home/sebastien/Desktop/Stample-react/src/repositories/reminderRepository.js", module);
```

Could it be related to this semicolon at the beginning?

**Comment by milankinen** [comment]
That semicolon should be correct JavaScript. Could you provide the codes causing this syntax error?

**Comment by slorber** [comment]
Hi,

Unfortunately it's a production app code it will be hard to share.


One weird thing I noticed was the fact that multiple reload requests were done while I just change the file once.

When I save a single file, it logs:

```
LiveReactload: new client connected
[10:50:06] Watchify detected change -> Rebuilding bundle
LiveReactload: sending reload request to 1 clients
LiveReactload: sending reload request to 1 clients
LiveReactload: sending reload request to 1 clients
LiveReactload: sending reload request to 1 clients
LiveReactload: sending reload request to 1 clients
LiveReactload: sending reload request to 1 clients
LiveReactload: sending reload request to 1 clients
[10:50:09] app.js was reloaded.
LiveReactload: sending reload request to 1 clients
```

I don't know what's the problem but it is definitely related to my issue.
I tried to debounce the reload requests and I don't get any error anymore.

```

  var notifyServer = http.createServer(_.debounce(function(req, res) {
    var body = 'ok';
    res.writeHead(200, {"Content-Length": body.length, "Content-Type": "text/plain"});
    res.end(body);

    log('sending reload request to ' + reloadServer.clients.length + ' clients');
    console.trace();
    reloadServer.clients.forEach(function(client) {
      client.send('plz reload');
    });
  },2000,false));
```

With this code, I get no parse error. I guess maybe the frontend app is trying to reload while the app.js file is being produced or something like that?

**Comment by milankinen** [comment]
I have a guess that Browserify writes the bundle in chunks. Each chuck causes file change event when *still writing* the file -> `SyntaxError`. This only shows up with big (usually production) bundles: only part of the bundle has been written when reloading occurs. 

Debouncing notify events is one possible solution but it causes a delay which distracts the development. 

I'll add JavaScript syntax check before reload event sending to prevent unwanted syntax errors.

**Comment by milankinen** [comment]
Should be fix'd in the newest version `0.6.0`.

**Comment by milankinen** [comment]
Not relevant since `2.x.x` releases. Closing this one
